Library/Homebrew/test/test_integration_cmds.rb

```diff
@@ -797,4 +797,41 @@ def test_custom_command
         cmd(cmd, {"PATH" => "#{path}#{File::PATH_SEPARATOR}#{ENV["PATH"]}"})
     end
   end
+
+  def test_search
+    formula_file = CoreTap.new.formula_dir/"testball.rb"
+    formula_file.write <<-EOS.undent
+      class Testball < Formula
+        desc "Some test"
+        url "https://example.com/testball-0.1.tar.gz"
+      end
+    EOS
+
+    desc_cache = HOMEBREW_CACHE.join("desc_cache.json")
+    refute_predicate desc_cache, :exist?, "Cached file should not exist"
+
+    assert_match "testball", cmd("search")
+    assert_match "testball", cmd("search", "testball")
+    assert_match "testball", cmd("search", "homebrew/homebrew-core/testball")
+    assert_match "testball", cmd("search", "--desc", "Some test")
+
+    flags = {
+      "macports" => "https://www.macports.org/ports.php?by=name&substr=testball",
+      "fink" => "http://pdb.finkproject.org/pdb/browse.php?summary=testball",
+      "debian" => "https://packages.debian.org/search?keywords=testball&searchon=names&suite=all&section=all",
+      "opensuse" => "https://software.opensuse.org/search?q=testball",
+      "fedora" => "https://admin.fedoraproject.org/pkgdb/packages/%2Atestball%2A/",
+      "ubuntu" => "http://packages.ubuntu.com/search?keywords=testball&searchon=names&suite=all&section=all"
```

Comment:
Interesting. I was actually unaware that a trailing comma was allowable! I just read through [this discussion](https://github.com/bbatsov/ruby-style-guide/issues/76) and while I see the merits of using it, my slight preference is still to do without it (but maybe this will change someday)... anyway, I'll incorporate this.

I saw that @apjanke started this [doc](https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/Prose-Style-Guidelines.md) a few months ago; are there any plans to make a similar document for coding style to fill the gaps between the Ruby style guide and the Homebrew DSL... when time allows? 😄